src/ImageSharp/Processing/AffineTransformBuilder.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Six Labors and contributors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.Collections.Generic;
+using System.Numerics;
+using SixLabors.ImageSharp.Processing.Processors.Transforms;
+using SixLabors.Primitives;
+
+namespace SixLabors.ImageSharp.Processing
+{
+    /// <summary>
+    /// A helper class for constructing <see cref="Matrix3x2"/> instances for use in affine transforms.
+    /// </summary>
+    public class AffineTransformBuilder
+    {
+        private readonly List<Matrix3x2> matrices = new List<Matrix3x2>();
+        private Rectangle rectangle;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AffineTransformBuilder"/> class.
+        /// </summary>
+        /// <param name="sourceSize">The source image size.</param>
+        public AffineTransformBuilder(Size sourceSize)
+        {
+            Guard.MustBeGreaterThan(sourceSize.Width, 0, nameof(sourceSize));
+            Guard.MustBeGreaterThan(sourceSize.Height, 0, nameof(sourceSize));
+
+            this.Size = sourceSize;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AffineTransformBuilder"/> class.
+        /// </summary>
+        /// <param name="sourceRectangle">The source rectangle.</param>
+        public AffineTransformBuilder(Rectangle sourceRectangle)
+            : this(sourceRectangle.Size)
+            => this.rectangle = sourceRectangle;
+
+        /// <summary>
+        /// Gets the source image size.
+        /// </summary>
+        internal Size Size { get; }
+
+        /// <summary>
+        /// Prepends a centered rotation matrix using the given rotation in degrees.
+        /// </summary>
+        /// <param name="degrees">The amount of rotation, in degrees.</param>
+        /// <returns>The <see cref="AffineTransformBuilder"/>.</returns>
+        public AffineTransformBuilder PrependRotateMatrixDegrees(float degrees)
```

Comment:
Having the word `Matrix` in the name of these methods seems unnecessarily verbose to me.
We can use it as `PrependRotationDegrees(...)`

We also need a `PrependRotationRadians(...)`.